### ISSUE TITLE
Element doesn't work for a list of lists, probably maps etc!

### ISSUE THREAD

**Issue by dennybaa** [hashicorp/terraform#8031]
Hello,

terraform 0.7.0

```
variable "v" {
    default = 0
}

variable "l2" {
    default = [1,2,3]
}

output "list-of-lists" {
    value = "${list(list(var.v), var.l2)}"
}

output "first-list-element-doesnt-work" {
    value = ["${element(list(list(var.v), var.l2), 0)}"]
}
```

It is what it is:

```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

list-of-lists = [
    [
        0
    ],
    [
        1,
        2,
        3
    ]
]
```

**Comment by maxstr** [comment]
+1 on this.

```
* module.my-node.aws_instance.node[1]: At column 3, line 1: concat: argument 1 should be type list, got type string in:

${concat(list("test"), element(list(list("test")), count.index))}
```

Has anyone found a workaround? My use case is that I'd like to pass as a variable a list of lists indexed by count to append to some default list.

**Comment by apparentlymart** [comment]
Hi @dennybaa! Sorry for the long silence here.

I just tried this against the v0.12.0-alpha2 build, exactly as you gave in your example, and confirmed that this is now working properly in master:

```
$ terraform apply

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

first-list-element-doesnt-work = [
  [
    0,
  ],
]
list-of-lists = [
  [
    0,
  ],
  [
    1,
    2,
    3,
  ],
]
```

The new configuration language implementation also allows for a more readable form for this expression, with the same result:

```hcl
output "list-of-lists" {
  value = [[var.v], var.l2]
}

output "first-list-element-doesnt-work" {
  value = [[var.v], var.l2][0]
}
```

This will be included in the forthcoming v0.12.0 release. Since the change is already in master, I'm going to close this out now. Thanks for reporting this, and sorry for the long delay in replying.
